tests/cocotb_tests/test_ral_read_write/test.py

```diff
@@ -51,38 +67,42 @@ def __init__(self, name):
         super().__init__(name)
         # do not use create map if only the default one
         # is intended to be used
-        self.def_map = uvm_reg_map("map")
-        self.def_map.configure(self, 0)
         self.CMD = command_reg("CMD")
-        self.CMD.configure(self, "0x100c", "", False, False)
-        self.def_map.add_reg(self.CMD, "0x0", "RW")
         self.ST = status_reg("ST")
-        self.ST.configure(self, "0x1014", "", False, False)
-        self.def_map.add_reg(self.ST, "0x0", "RO")
+        self.CMD.configure(self)
+        self.ST.configure(self)
+        self.def_map = self.create_map(
+            "map", 0x0, 4, uvm_endianness_e.UVM_LITTLE_ENDIAN, True
+        )
+        self.def_map.add_reg(self.CMD, 0x100C, "RW")
+        self.def_map.add_reg(self.ST, 0x1014, "RO")
+        self.lock_model()
 
 
 # UVM Sequence item class
 class cmd_item(uvm_sequence_item):
     def __init__(self, name):
         super().__init__(name)
         self.Endian = False
-        self.CMD_data = BinaryValue(0, 32, self.Endian)
-        self.WR_RD = BinaryValue(0, 1, self.Endian)
+        self.CMD_data = LogicArray(0, 32)
+        self.WR_RD = LogicArray(0, 1)
         self.write_data: int = 0
         self.read_data: int = 0
-        self.addr: str = ""
+        self.addr: int = None
```

Comment:
Either remove type annotations or use them correctly and check them. We would avoid implicit optionals.